### ISSUE TITLE
loss function doesn't ignore large masked values

### ISSUE THREAD

**Issue by ssfrr** [fchollet/keras#6825]
```python
import numpy as np
from keras.models import Model
from keras.layers import Input, Masking

maskval = 1e20
l1 = Input(shape=(10, 4))
l2 = Masking(mask_value=maskval)(l1)

m = Model(inputs=l1, outputs=l2)
m.compile(optimizer='sgd', loss='mse')

x = np.random.rand(6, 10, 4)
y = np.copy(x)
x[1, 3:, :] = maskval
x[3, 5:, :] = maskval
x[5, 7:, :] = maskval

y += 1
y[1, 3:, :] = maskval
y[3, 5:, :] = maskval
y[5, 7:, :] = maskval

m.test_on_batch(x, y) # returns nan
```

If you comment out the `y[...] = maskval` lines, or change `maskval` to a smaller number then it works fine (MSE loss is 1.0).
